internal/hncconfig/validator.go

```diff
@@ -96,31 +97,38 @@ func (v *Validator) handle(ctx context.Context, inst *api.HNCConfiguration) admi
 }
 
 func (v *Validator) validateTypes(inst *api.HNCConfiguration, ts gvkSet) admission.Response {
+	allErrs := field.ErrorList{}
 	for i, r := range inst.Spec.Resources {
 		gr := schema.GroupResource{Group: r.Group, Resource: r.Resource}
-		field := field.NewPath("spec", "resources").Index(i)
+		fldPath := field.NewPath("spec", "resources").Index(i)
 		// Validate the type configured is not an HNC enforced type.
 		if api.IsEnforcedType(r) {
-			return webhooks.DenyInvalid(field, fmt.Sprintf("Invalid configuration of %s in the spec, because it's enforced by HNC "+
-				"with 'Propagate' mode. Please remove it from the spec.", gr))
+			fldErr := field.Invalid(fldPath, gr, "enforced by HNC with 'Propagate' mode. Please remove it from the spec.")
+			allErrs = append(allErrs, fldErr)
 		}
 
 		// Validate the type exists in the apiserver. If yes, convert GR to GVK. We
 		// use GVK because we will need to checkForest() later to avoid source
 		// overwriting conflict (forest uses GVK as the key for object reconcilers).
 		gvk, err := v.translator.gvkForGR(gr)
 		if err != nil {
-			return webhooks.DenyInvalid(field,
-				fmt.Sprintf("Cannot find the %s in the apiserver with error: %s", gr, err.Error()))
+			fldErr := field.Invalid(fldPath, gr, fmt.Sprintf("not found in the apiserver with error: %s", err.Error()))
```

Comment:
Better: "unknown resource: %s"

Reply:
The field value is added to the message by field.Invalid, but I modified it according to your suggestion. Example from unit-tests:
````
HNCConfiguration.hnc.x-k8s.io \"config\" is invalid: spec.resources[0]: Invalid value: crontabs: unknown resource
````